spring-rabbit/src/main/java/org/springframework/amqp/rabbit/retry/MissingMessageIdAdvice.java

```diff
@@ -40,8 +40,11 @@
  * Must be placed before the retry interceptor in the advice chain.
  * @author Gary Russell
  * @since 1.1.2
+ * @deprecated since 1.7 in favor of non-retryable messages logic in the {@code StatefulRetryOperationsInterceptor}
  *
  */
+@Deprecated
+@SuppressWarnings("deprecation")
```

Comment:
Eh... As for me the last variant sounds better as a smooth movement from 1.6 to 2.0 over 1.7.

Although I admit that I don't understand this new Retry logic yet...